python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -325,48 +259,13 @@ def get_nodes_to_launch(
         logger.debug("Node requests: {}".format(total_nodes_to_add))
         return total_nodes_to_add, final_unfulfilled
 
-    def _legacy_worker_node_to_launch(
-        self,
-        nodes: List[NodeID],
-        launching_nodes: Dict[NodeType, int],
-        node_resources: List[ResourceDict],
-        resource_demands: List[ResourceDict],
-    ) -> Dict[NodeType, int]:
-        """Get worker nodes to launch when resources missing in legacy yamls.
-
-        If there is unfulfilled demand and we don't know the resources of the
-        workers, it returns max(1, min_workers) worker nodes from which we
-        later calculate the node resources.
-        """
-        # Populate worker list.
-        _, worker_nodes = self._get_head_and_workers(nodes)
-
-        if self.max_workers == 0:
-            return {}
-        elif sum(launching_nodes.values()) + len(worker_nodes) > 0:
-            # If we are already launching a worker node, wait for its resources
-            # to be known.
-            # TODO(ameer): Note that if first worker node fails this will never
-            # launch any more nodes.
-            return {}
-        else:
-            unfulfilled, _ = get_bin_pack_residual(node_resources, resource_demands)
-            workers_to_add = min(
-                self.node_types[NODE_TYPE_LEGACY_WORKER].get("min_workers", 0),
-                self.node_types[NODE_TYPE_LEGACY_WORKER].get("max_workers", 0),
-            )
-            if workers_to_add > 0 or unfulfilled:
-                return {NODE_TYPE_LEGACY_WORKER: max(1, workers_to_add)}
-            else:
-                return {}
-
     def _update_node_resources_from_runtime(
         self, nodes: List[NodeID], max_resources_by_ip: Dict[NodeIP, ResourceDict]
     ):
         """Update static node type resources with runtime resources
 
         This will update the cached static node type resources with the runtime
-        resources. Because we can not know the correctly memory or
+        resources. Because we can not know the exact autofilled memory or
```

Comment:
Maybe something like

`This is needed because we cannot use the autoscaling config file to determine the exact memory or object store memory available at runtime.`